darts/models/forecasting/forecasting_model.py

```diff
@@ -444,105 +696,202 @@ def historical_forecasts(
             inspect.signature(retrain_func).parameters.keys()
         )
 
-        # prepare the start parameter -> pd.Timestamp
-        start = series.get_timestamp_at_point(start)
+        series = series2seq(series)
+        past_covariates = series2seq(past_covariates)
+        future_covariates = series2seq(future_covariates)
 
-        # build the prediction times in advance (to be able to use tqdm)
-        last_valid_pred_time = self._get_last_prediction_time(
-            series, forecast_horizon, overlap_end
-        )
+        # If the model has never been fitted before using historical_forecasts,
+        # we need to know if it uses past or future covariates. The only possible assumption is that
+        # the user is using the same covariates as he would in the fit method.
```

Comment:
:) 
```suggestion
        # the user is using the same covariates as they would in the fit method.
```

Reply:
it actually what I meant, as  if `the user would have made the same choice if he had used fit before historical_forecasts()`

Reply:
I think I understand your sentence, but that was an attempt to make it gender neutral as not all our users are male :D